contrib/mpi-proxy-split/lower-half/libproxy.c

```diff
@@ -110,7 +110,7 @@ getTextSegmentRange(pid_t proc,                 // IN
   if (proc == -1) {
     f = fopen("/proc/self/stat", "r");
   } else {
-    char pids[] = "/proc/XXXXXX/stat";
+    char pids[] = "/proc/XXXXXXX/stat";
```

Comment:
Please add a comment:

On 64-bit systems, pid_max can be set to any value up to 2^22 (PID_MAX_LIMIT, approximately 4 million)

NIT: I would just use `char pids[PATH_MAX]`

Reply:
both ideas sound good..